### PR TITLE
refactor(batch-details): move logic into existing cmd

### PR DESCRIPTION
**Pull request by fborello-lambda**
# Purpose
- Move the logic of `zks prover batch-details` into `zks chain l1-batch-details`

## Help msg

```
❯ zks chain l1-batch-details --help
Retrieves details for a given L1 batch.

Usage: zks chain l1-batch-details [OPTIONS]

Options:
  -n <BATCHES>...      
  -t                   The command displays the proof-time based on the L1 txs' timestamps.
  -h, --help           Print help
```

## Example

```
❯ zks chain l1-batch-details -t -n 55
✓ Success                         
======== Batch 00055 Status ========
Committed At: 2024-09-19 01:18:50.224044 UTC
Commit Tx Hash: 0xd1de5efc3b71b8538328f80b750b95fe3a0d9b887a00eb77c898da66dbf0aea9
Proven At: 2024-09-19 15:54:37.330386 UTC
Proven Tx Hash: 0xcbbaaeafecd3e12bf3fc64d54560110d34459c03689ba4f92ccd0eac301db0ad
ProofTime from Committed to Proven: 14:35:47
```

### CODE REVIEW

File: src/commands/chain.rs [hunk 1]
```diff
@@ -36,7 +37,16 @@ pub(crate) enum Command {
         limit: u8,
     },
     #[clap(about = "Retrieves details for a given L1 batch.")]
-    L1BatchDetails { batch: u32 },
+    L1BatchDetails {
+        #[clap(short = 'n', num_args = 1..)]
+        batches: Option<Vec<L1BatchNumber>>,
```
Comment (ilitteri) [inline src/commands/chain.rs]: Let's try this as an `Option<Vec<T>>` can be treated as an empty `Vec<T>`, if `--batches` is not passed, then the default will be an empty `Vec`.
```suggestion
        #[clap(short = 'n', num_args = 1.., default_value_t = Vec::default)]
        batches: Option<Vec<L1BatchNumber>>,
```

Comment (ilitteri) [reply src/commands/chain.rs]: Then the logic below will be simplified.

File: src/commands/chain.rs [hunk 2]
```diff
@@ -218,9 +228,28 @@ impl Command {
...
+                if proof_time {
+                    display_batches_proof_time_from_l1_batch_details(
+                        batches_vec,
+                        current_batch,
+                        l2_provider,
+                    )
+                    .await?;
+                } else {
+                    display_batches_details(batches_vec, current_batch, l2_provider).await?;
+                }
```
Comment (ilitteri) [inline src/commands/chain.rs]: If you agree with the above these should be the new changes (untested).

```suggestion
            Command::L1BatchDetails {
                mut batches,
                proof_time,
            } => {
                let current_batch = l2_provider.get_l1_batch_number().await?.as_u32().into();

                if batches.is_empty() {
                    batches.push(current_batch);
                }

                if proof_time {
                    display_batches_proof_time_from_l1_batch_details(
                        batches,
                        current_batch,
                        l2_provider,
                    )
                    .await?;
                } else {
                    display_batches_details(batches, current_batch, l2_provider).await?;
                }
```

**Review by ilitteri — APPROVED** [review]
LGTM, only one suggestion

**Comment by fborello-lambda** [comment]
I've just integrated the suggested changes. The `default_value_t = Vec::default()` is not needed, and it cannot be used because `L1BatchNumber` does not implement the `ToString` trait.

**Pull request merged**